test/jest/cookieSyncManager.spec.ts

```diff
@@ -0,0 +1,454 @@
+
+            expect(cookieSyncManager.performCookieSync).toHaveBeenCalledWith(
+                '',
+                5,
+                testMPID,
+                {},
+                undefined,
+                true,
+                false, 
+            );
+        });
+
+        it('should toggle requiresConsent value if filtering filteringConsentRuleValues are defined', () => {
+            const myPixelSettings: PixelConfiguration = {
+                filteringConsentRuleValues: {
+                    values: ['test'],
+                },
+            };
+
+            const mockSDK = ({
+                _Store: {
+                    webviewBridgeEnabled: false,
+                    pixelConfigurations: [{...pixelSettings, ...myPixelSettings}],
+                },
+                _Persistence: {
+                    getPersistence: () => ({testMPID: {
+                        csd: {}
+                    }}),
+                },
+            } as unknown) as MParticleWebSDK;
+
+            const cookieSyncManager = new CookieSyncManager(mockSDK);
+            cookieSyncManager.performCookieSync = jest.fn();
+
+            cookieSyncManager.attemptCookieSync(null, testMPID, true);
+
+            expect(cookieSyncManager.performCookieSync).toHaveBeenCalledWith(
+                '',
+                5,
+                testMPID,
+                {},
+                { values: ['test'] },
+                true,
+                true, 
+            );
+        });
+
+        it('should update the urlWithRedirect if a redirectUrl is defined', () => {
+            const myPixelSettings: PixelConfiguration = {
+                pixelUrl: 'https://test.com',
+                redirectUrl: '?redirect=https://redirect.com&mpid=%%mpid%%',
+            };
+
+            const mockSDK = ({
+                _Store: {
+                    webviewBridgeEnabled: false,
+                    pixelConfigurations: [{...pixelSettings, ...myPixelSettings}],
+                },
+                _Persistence: {
+                    getPersistence: () => ({testMPID: {
+                        csd: {}
+                    }}),
+                },
+            } as unknown) as MParticleWebSDK;
+
+            const cookieSyncManager = new CookieSyncManager(mockSDK);
+            cookieSyncManager.performCookieSync = jest.fn();
+
+            cookieSyncManager.attemptCookieSync(null, testMPID, true);
+
+            expect(cookieSyncManager.performCookieSync).toHaveBeenCalledWith(
+                'https://test.com%3Fredirect%3Dhttps%3A%2F%2Fredirect.com%26mpid%3DtestMPID',
+                5,
+                testMPID,
+                {},
+                undefined,
+                true,
+                false, 
+            );
+        });
+
+        // QUESTION: What is the purpose of this code path?
+        it('should call performCookieSync with mpid if previousMpid and mpid do not match', () => {
```

Comment:
If a cookie sync was recently done, it won't be performed again until a certain time interval (i thinkt his varies per partner).  So if MPID1 is refreshing the page, it will do a cookie sync once until a certain time passes.

If a user logs out, and then logs in, then we would immediately know it's a different user and not have to use that logic since the new user hasn't done a cookie sync yet.

As I write that all out, I realize there is room for improvement.  There is a chance that if MPID1 logs in, cookie syncs, then logs in as MPID2, and cookie syncs, then logs in again as MPID1, i can see it wouldn't consider the time and ONLY looks at MPID change before performing a cookie sync.  Can you make a ticket to address this scenario?  It's a very minor optimization.